### PR TITLE
Completed draft: Trip Organizer

### PR DESCRIPTION
**Pull request by tboonma**
workflow
codecov

### What is the feature?
- Trip organizer feature that help people order their places in a trip.
- Shows estimated arrival time for each place in a trip.
### What is the solution?
- Using Google Map JavaScript API.
- Google Direction API called in Backend (Django).
### What areas of the site does it impact?
- Added navigation link to this function in Navigation bar.
### How to test?
- Click `Planner` in navigation bar and select amount days spent in a trip and search places to add to the trip. You can also reorder places in a trip by the buttons on the right. After added all places, click `Calculate Traveling Time` to get estimated arrival time.
### Screen shot
Screen Shot 2564-11-15 at 01 50 29

### CODE REVIEW

File: trip/views.py [hunk 1]
```diff
@@ -212,6 +211,56 @@ def place_info(request, place_id: str):
...
+
+
+@login_required
+def get_travel_time(request) -> JsonResponse:
+    """Get How long does it takes between places receiving POST method as a list of place id.
+
+    POST params:
+        places: list of places that will be calculated the direction ordered by items order in the list.
+    Returns:
+        JsonResponse: all data about direction from origin to destination.
```
Comment (nabhan-au) [inline trip/views.py:251]: Redundant dostring

**Review by nabhan-au — APPROVED** [review]
Nice work!!

**Review by TaninDean — COMMENTED** [review]
Good job!

**Review by TaninDean — APPROVED** [review]

**Comment by tboonma** [comment]
Thank you

**Pull request merged**